epochStart/bootstrap/process.go

```diff
@@ -1206,11 +1207,35 @@ func (e *epochStartBootstrap) updateDataForScheduled(
 		return nil, err
 	}
 
-	res.rootHashToSync = e.dataSyncerWithScheduled.GetRootHashToSync(shardNotarizedHeader)
+	rootHashToSync := e.dataSyncerWithScheduled.GetRootHashToSync(shardNotarizedHeader)
+
+	if shardNotarizedHeader.IsHeaderV3() {
+		rootHashToSync = e.getRootHashToSync(shardNotarizedHeader)
+	}
+
+	res.rootHashToSync = rootHashToSync
 
 	return res, nil
 }
 
+func (e *epochStartBootstrap) getRootHashToSync(
+	header data.HeaderHandler,
+) []byte {
+	rootHashToSync := header.GetRootHash()
```

Comment:
this is not correct if there are scheduled transactions, should call e.dataSyncerWithScheduled.GetRootHashToSync
so maybe this function can include also line 1210, and remove it from there?

Also metachain does not have scheduled roothash

Reply:
right, missed it